### PR TITLE
Add UT for pkg/controllers/binding/binding_controller.go

### PR DESCRIPTION
**Pull request by zishen**
Signed-off-by: zishen <softwarebtg@163.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Promote UT coverage

**Which issue(s) this PR fixes**:
Part of #3327

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~

File: pkg/controllers/binding/binding_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tempDyClient := fakedynamic.NewSimpleDynamicClient(gclient.NewSchema())
+			c := &ResourceBindingController{
+				Client:          fake.NewClientBuilder().WithScheme(gclient.NewSchema()).Build(),
+				RESTMapper:      testing2.RestMapper,
```
[the remark this answers is not in the document]
Comment (zishen) [reply pkg/controllers/binding/binding_controller_test.go]: > How about constructing the `RESTMapper` in a public package?

Good idea. How about test/helper package? And new file named "manager.go"?

File: pkg/controllers/binding/binding_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+				InformerManager: genericmanager.NewSingleClusterInformerManager(tempDyClient, 0, nil),
+				DynamicClient:   tempDyClient,
+			}
+			if tt.existingRB {
```
[the remark this answers is not in the document]
Comment (zishen) [reply pkg/controllers/binding/binding_controller_test.go]: > Here can use `tt.rb != nil` to replace, so we can emit the input `existingRB` param.

OK,It is indeed necessary.

File: pkg/controllers/binding/binding_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+				// Add a rb to the fake client.
+				if err := c.Client.Create(context.Background(), tt.rb); err != nil {
+					t.Fatalf("Failed to create rb: %v", err)
+				}
+			}
+			// Run the reconcile function.
+			req := controllerruntime.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "test-rb",
+					Namespace: "default",
```
[the remark this answers is not in the document]
Comment (zishen) [reply pkg/controllers/binding/binding_controller_test.go]: > How to construct the namespace and name as input parameters?
yes, I have make all req strcuct as input parameters.  Thanks

File: pkg/controllers/binding/binding_controller_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (XiShanYongYe-Chang) [reply pkg/controllers/binding/binding_controller_test.go]: I think it is ok.

**Comment by XiShanYongYe-Chang** [comment]
Hi @zishen, can you help squash your commit into one, and then we can ask @RainbowMango to help approve to run the ci test.

**Comment by zishen** [comment]
> Hi @zishen, can you help squash your commit into one, and then we can ask @RainbowMango to help approve to run the ci test.

ok. I have done it. @RainbowMango help to approve

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks
/lgtm

**Pull request merged**